### PR TITLE
[Feature] 346981 - Add design system buttons

### PR DESCRIPTION
**Pull request by mpamaro**
- [Lib] Add new disabled background color
- [Lib] Change DSButtonText and his style so that the color property becomes required
- [Lib] Add new primary button widget
- [Lib] Add new secondary button widget
- [Lib] Add new tertiary button widget
- [Lib] Add new DSButton to be a "container" for primary, secondary and tertiary buttons
- [Lib] Add new ring-style loading widget
- [Lib] Change the name of existing loading widget to have the "Fading" term
- [Lib] Add animation to the new buttons
- [Lib] Remove the suffix Widget from some previously created widgets
- [Lib] Fix buttons text overflow
- [Lib] Set minimum size for buttons
- [Lib] Add default duration for animations in DSUtils
- [Lib] Add animation to DSMessageBubble
- [Lib] Suppress onPressed if button is disabled or loading
- [Lib] Add property to set that the bubble should be grouping with the previous bubble
- [Lib] Add widget documentation to some classes that doesn't have one
- [Lib] Add font variant in texts documentation
- [Lib] Change CHANGELOG file to include new features
- [Lib] Change folder of loading files
- [Lib] Fix CachedNetworkImage uncaught exception when URL is empty
- [Sample] Add group of message bubble
- [Sample]  Add button showcase
- [Sample]  Change all showcase files name to have the .showcase suffix.

### CODE REVIEW

**Review by githubdoandre — APPROVED** [review]

**Review by jwdutra — APPROVED** [review]

File: lib/src/widgets/chat/ds_message_bubble.widget.dart [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
   const DSMessageBubble({
     Key? key,
     required this.align,
     required this.child,
     this.borderRadius = const [DSBorderRadius.all],
-    this.contentPadding = const EdgeInsets.symmetric(
-      vertical: 10.0,
-      horizontal: 14.0,
+    this.padding = const EdgeInsets.symmetric(
+      vertical: 10,
```
Comment (leonardogbr) [inline lib/src/widgets/chat/ds_message_bubble.widget.dart:20]: It'll conflict with Andre's commit. The correct is `vertical: 8` and `horizontal: 16`. You can accept the incoming change

Comment (mpamaro) [reply lib/src/widgets/chat/ds_message_bubble.widget.dart:20]: Sure, and there are even more files that will conflict. That's why I decided to be the last to merge. Anyway, thanks for the advice.

**Review by leonardogbr — APPROVED** [review]

**Review by githubdoandre — APPROVED** [review]

File: lib/src/widgets/buttons/ds_button.widget.dart [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    this.trailingIcon,
+    this.borderColor,
+    this.disable = false,
+    this.loading = false,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return OutlinedButton(
+      onPressed: !disable || !loading ? onPressed : null,
```
Comment (leonardogbr) [inline lib/src/widgets/buttons/ds_button.widget.dart:38]: I'm using the disable property, but the onPressed function keeps getting called. Maybe you should change the OR operator to an AND operator.

File: lib/src/widgets/buttons/ds_button.widget.dart [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+          44.0,
+        ),
+        primary: foregroundColor,
+        backgroundColor: backgroundColor,
+        shape: RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(8.0),
+        ),
+        side: BorderSide(
+          color: borderColor ?? backgroundColor,
+        ),
```
Comment (leonardogbr) [inline lib/src/widgets/buttons/ds_button.widget.dart:51]: You have to control the padding for different scenarios. When it's icon only, it have a different padding

**Review by leonardogbr — CHANGES_REQUESTED** [review]

**Review by leonardogbr — APPROVED** [review]

**Pull request merged**